src/helpers.rs

```diff
@@ -827,6 +881,123 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             None => tcx.item_name(def_id),
         }
     }
+
+    // extract the scalar value from the result of reading a scalar from the machine
+    // and convert it to a CArg
+    fn extract_scalar(
+        scalar_res: InterpResult<'tcx, ScalarMaybeUninit<Tag>>,
```

Comment:
I think you should pass in just a `ScalarMaybeUninit` and `?` away the `InterpResult` at the call sites. The unsupported argument message is misleading imo.

Reply:
good idea